packages/core/src/sequelize.js

```diff
@@ -743,25 +743,23 @@ Use Sequelize#query if you wish to use replacements.`);
    * Execute a query which would set an environment or user variable. The variables are set per connection, so this function needs a transaction.
    * Only works for MySQL or MariaDB.
    *
-   * @param {object}        variables Object with multiple variables.
-   * @param {object}        [options] query options.
-   * @param {Transaction}   [options.transaction] The transaction that the query should be executed under
-   *
-   * @memberof Sequelize
+   * @param {object} variables Object with multiple variables.
+   * @param {object} [options] query options.
    *
    * @returns {Promise}
    */
-  async set(variables, options) {
-
+  async setSessionVariables(variables, options) {
     // Prepare options
-    options = { ...this.options.set, ...typeof options === 'object' && options };
+    options = { ...this.options.set, ...options };
```

Comment:
Do we want to keep using `options.set` or do we also want to rename it to something more descriptive?

Reply:
It should be aligned with the name of the method so setSessionVariables